lib/helpers/package-json.js

```diff
@@ -30,10 +30,10 @@ function setNewValue(currentFileContent, depType, depName, newVersion) {
     return currentFileContent;
   }
   // Update the file = this is what we want
-  parsedContents[depType][depName] = newVersion;
+  parsedContents[depType][depName] = (upgradeType === 'major' ? '^' : upgradeType === 'minor' ? '~' : '') + newVersion;
```

Comment:
There's a major flaw in my approach, as I can't reasonable decide which range character to use based on the `upgradeType`.

Even a pinned version may require a major upgrade.

Therefore I'll need to extract the original range character and then prepend the new version with that range.